command_line_assistant/utils/files.py

```diff
@@ -93,3 +100,94 @@ def write_file(contents: Union[str, bytes], path: Path, mode: int = 0o600) -> No
             path,
             str(e),
         )
+
+
+class NamedFileLock:
+    """File-based lock mechanism for general locking purposes.
+
+    This class provides a way to track state across processes
+    using a lock file. The lock file contains the PID of the capturing process.
+
+    Usage:
+        with NamedFileLock() as lock:
+            # Execute your operations
+        # Lock is automatically released
+    """
+
+    def __init__(self, name: str):
+        """Initialize the named file lock mechanism."""
+        self._pid = os.getpid()
+        self._name = name
+        self._lock_file = Path(get_xdg_state_path(), name + ".lock")
+
+    @property
+    def is_locked(self) -> bool:
+        """Check if a lock is currently active.
+
+        This checks if the lock file exists and if the PID in it is still running.
+
+        Returns:
+            bool: True if a lock is active, False otherwise
+        """
+        if not self._lock_file.exists():
+            return False
+
+        try:
+            pid = int(self._lock_file.read_text().strip())
+            # Check if process is still running
+            os.kill(pid, 0)
```

Comment:
I find this unusual. Calling a function named "kill" to check if a process is alive. ^_^

Reply:
Yeah, it's weird. But do you know a different way? I guess there's psutil that can check for that, but I'm not sure about pulling a new dependency just for this

Reply:
Unfortunately, no. I spent some time reading over the `kill` function and `signals` package, and nothing more obvious stands out to me.